File: androidApp/src/main/java/nl/codeface/letsee_kmm/android/Breed.kt
package nl.codeface.letsee_kmm.android

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Breed(
    val id: String,
    val name: String,
    val description: String = "",
    val origin: String = "",
    val temperament: String = "",
    @SerialName("wikipedia_url") val wikipediaUrl: String? = null,
    @SerialName("life_span") val lifeSpan: String? = null,
    val weight: Weight? = null,
    val image: BreedImage? = null,
)

@Serializable
data class Weight(
    val imperial: String? = null,
    val metric: String? = null,
)

@Serializable
data class BreedImage(
    val id: String? = null,
    val url: String? = null,
)

File: iosApp/iosApp/Breed.swift
import Foundation

struct Breed: Codable, Identifiable {
    let id: String
    let name: String
    let description: String
    let origin: String
    let temperament: String
    let wikipedia_url: String?
    let life_span: String?
    let weight: Weight?
    let image: BreedImage?
}

struct Weight: Codable {
    let imperial: String?
    let metric: String?
}

struct BreedImage: Codable {
    let id: String?
    let url: String?
}
